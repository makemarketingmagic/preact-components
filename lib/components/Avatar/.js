!function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:!1,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=!0,o.exports}n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},n.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&"object"==typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var o in e)n.d(r,o,function(t){return e[t]}.bind(null,o));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,"a",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p="/",n(n.s="./components/Avatar/index.js")}({"../node_modules/classnames/index.js":function(e,t,n){var r;
/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
*/
/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
*/
!function(){"use strict";var n={}.hasOwnProperty;function o(){for(var e=[],t=0;t<arguments.length;t++){var r=arguments[t];if(r){var a=typeof r;if("string"===a||"number"===a)e.push(r);else if(Array.isArray(r)&&r.length){var i=o.apply(null,r);i&&e.push(i)}else if("object"===a)for(var l in r)n.call(r,l)&&r[l]&&e.push(l)}}return e.join(" ")}void 0!==e&&e.exports?(o.default=o,e.exports=o):void 0===(r=function(){return o}.apply(t,[]))||(e.exports=r)}()},"../node_modules/preact/dist/preact.esm.js":function(e,t,n){"use strict";n.r(t),n.d(t,"h",function(){return i}),n.d(t,"createElement",function(){return i}),n.d(t,"cloneElement",function(){return s}),n.d(t,"Component",function(){return B}),n.d(t,"render",function(){return L}),n.d(t,"rerender",function(){return d}),n.d(t,"options",function(){return r});var r={},o=[],a=[];function i(e,t){var n,i,l,u,s=a;for(u=arguments.length;u-- >2;)o.push(arguments[u]);for(t&&null!=t.children&&(o.length||o.push(t.children),delete t.children);o.length;)if((i=o.pop())&&void 0!==i.pop)for(u=i.length;u--;)o.push(i[u]);else"boolean"==typeof i&&(i=null),(l="function"!=typeof e)&&(null==i?i="":"number"==typeof i?i=String(i):"string"!=typeof i&&(l=!1)),l&&n?s[s.length-1]+=i:s===a?s=[i]:s.push(i),n=l;var c=new function(){};return c.nodeName=e,c.children=s,c.attributes=null==t?void 0:t,c.key=null==t?void 0:t.key,void 0!==r.vnode&&r.vnode(c),c}function l(e,t){for(var n in t)e[n]=t[n];return e}var u="function"==typeof Promise?Promise.resolve().then.bind(Promise.resolve()):setTimeout;function s(e,t){return i(e.nodeName,l(l({},e.attributes),t),arguments.length>2?[].slice.call(arguments,2):e.children)}var c=/acit|ex(?:s|g|n|p|$)|rph|ows|mnc|ntw|ine[ch]|zoo|^ord/i,p=[];function f(e){!e._dirty&&(e._dirty=!0)&&1==p.push(e)&&(r.debounceRendering||u)(d)}function d(){var e,t=p;for(p=[];e=t.pop();)e._dirty&&E(e)}function v(e,t,n){return"string"==typeof t||"number"==typeof t?void 0!==e.splitText:"string"==typeof t.nodeName?!e._componentConstructor&&m(e,t.nodeName):n||e._componentConstructor===t.nodeName}function m(e,t){return e.normalizedNodeName===t||e.nodeName.toLowerCase()===t.toLowerCase()}function _(e){var t=l({},e.attributes);t.children=e.children;var n=e.nodeName.defaultProps;if(void 0!==n)for(var r in n)void 0===t[r]&&(t[r]=n[r]);return t}function h(e){var t=e.parentNode;t&&t.removeChild(e)}function y(e,t,n,r,o){if("className"===t&&(t="class"),"key"===t);else if("ref"===t)n&&n(null),r&&r(e);else if("class"!==t||o)if("style"===t){if(r&&"string"!=typeof r&&"string"!=typeof n||(e.style.cssText=r||""),r&&"object"==typeof r){if("string"!=typeof n)for(var a in n)a in r||(e.style[a]="");for(var a in r)e.style[a]="number"==typeof r[a]&&!1===c.test(a)?r[a]+"px":r[a]}}else if("dangerouslySetInnerHTML"===t)r&&(e.innerHTML=r.__html||"");else if("o"==t[0]&&"n"==t[1]){var i=t!==(t=t.replace(/Capture$/,""));t=t.toLowerCase().substring(2),r?n||e.addEventListener(t,b,i):e.removeEventListener(t,b,i),(e._listeners||(e._listeners={}))[t]=r}else if("list"!==t&&"type"!==t&&!o&&t in e)!function(e,t,n){try{e[t]=n}catch(e){}}(e,t,null==r?"":r),null!=r&&!1!==r||e.removeAttribute(t);else{var l=o&&t!==(t=t.replace(/^xlink:?/,""));null==r||!1===r?l?e.removeAttributeNS("http://www.w3.org/1999/xlink",t.toLowerCase()):e.removeAttribute(t):"function"!=typeof r&&(l?e.setAttributeNS("http://www.w3.org/1999/xlink",t.toLowerCase(),r):e.setAttribute(t,r))}else e.className=r||""}function b(e){return this._listeners[e.type](r.event&&r.event(e)||e)}var g=[],x=0,C=!1,N=!1;function w(){for(var e;e=g.pop();)r.afterMount&&r.afterMount(e),e.componentDidMount&&e.componentDidMount()}function S(e,t,n,r,o,a){x++||(C=null!=o&&void 0!==o.ownerSVGElement,N=null!=e&&!("__preactattr_"in e));var i=k(e,t,n,r,a);return o&&i.parentNode!==o&&o.appendChild(i),--x||(N=!1,a||w()),i}function k(e,t,n,r,o){var a=e,i=C;if(null!=t&&"boolean"!=typeof t||(t=""),"string"==typeof t||"number"==typeof t)return e&&void 0!==e.splitText&&e.parentNode&&(!e._component||o)?e.nodeValue!=t&&(e.nodeValue=t):(a=document.createTextNode(t),e&&(e.parentNode&&e.parentNode.replaceChild(a,e),j(e,!0))),a.__preactattr_=!0,a;var l=t.nodeName;if("function"==typeof l)return function(e,t,n,r){var o=e&&e._component,a=o,i=e,l=o&&e._componentConstructor===t.nodeName,u=l,s=_(t);for(;o&&!u&&(o=o._parentComponent);)u=o.constructor===t.nodeName;o&&u&&(!r||o._component)?(A(o,s,3,n,r),e=o.base):(a&&!l&&(O(a),e=i=null),o=T(t.nodeName,s,n),e&&!o.nextBase&&(o.nextBase=e,i=null),A(o,s,1,n,r),e=o.base,i&&e!==i&&(i._component=null,j(i,!1)));return e}(e,t,n,r);if(C="svg"===l||"foreignObject"!==l&&C,l=String(l),(!e||!m(e,l))&&(a=function(e,t){var n=t?document.createElementNS("http://www.w3.org/2000/svg",e):document.createElement(e);return n.normalizedNodeName=e,n}(l,C),e)){for(;e.firstChild;)a.appendChild(e.firstChild);e.parentNode&&e.parentNode.replaceChild(a,e),j(e,!0)}var u=a.firstChild,s=a.__preactattr_,c=t.children;if(null==s){s=a.__preactattr_={};for(var p=a.attributes,f=p.length;f--;)s[p[f].name]=p[f].value}return!N&&c&&1===c.length&&"string"==typeof c[0]&&null!=u&&void 0!==u.splitText&&null==u.nextSibling?u.nodeValue!=c[0]&&(u.nodeValue=c[0]):(c&&c.length||null!=u)&&function(e,t,n,r,o){var a,i,l,u,s,c=e.childNodes,p=[],f={},d=0,m=0,_=c.length,y=0,b=t?t.length:0;if(0!==_)for(var g=0;g<_;g++){var x=c[g],C=x.__preactattr_,N=b&&C?x._component?x._component.__key:C.key:null;null!=N?(d++,f[N]=x):(C||(void 0!==x.splitText?!o||x.nodeValue.trim():o))&&(p[y++]=x)}if(0!==b)for(var g=0;g<b;g++){u=t[g],s=null;var N=u.key;if(null!=N)d&&void 0!==f[N]&&(s=f[N],f[N]=void 0,d--);else if(!s&&m<y)for(a=m;a<y;a++)if(void 0!==p[a]&&v(i=p[a],u,o)){s=i,p[a]=void 0,a===y-1&&y--,a===m&&m++;break}s=k(s,u,n,r),l=c[g],s&&s!==e&&s!==l&&(null==l?e.appendChild(s):s===l.nextSibling?h(l):e.insertBefore(s,l))}if(d)for(var g in f)void 0!==f[g]&&j(f[g],!1);for(;m<=y;)void 0!==(s=p[y--])&&j(s,!1)}(a,c,n,r,N||null!=s.dangerouslySetInnerHTML),function(e,t,n){var r;for(r in n)t&&null!=t[r]||null==n[r]||y(e,r,n[r],n[r]=void 0,C);for(r in t)"children"===r||"innerHTML"===r||r in n&&t[r]===("value"===r||"checked"===r?e[r]:n[r])||y(e,r,n[r],n[r]=t[r],C)}(a,t.attributes,s),C=i,a}function j(e,t){var n=e._component;n?O(n):(null!=e.__preactattr_&&e.__preactattr_.ref&&e.__preactattr_.ref(null),!1!==t&&null!=e.__preactattr_||h(e),P(e))}function P(e){for(e=e.lastChild;e;){var t=e.previousSibling;j(e,!0),e=t}}var M={};function T(e,t,n){var r,o=M[e.name];if(e.prototype&&e.prototype.render?(r=new e(t,n),B.call(r,t,n)):((r=new B(t,n)).constructor=e,r.render=U),o)for(var a=o.length;a--;)if(o[a].constructor===e){r.nextBase=o[a].nextBase,o.splice(a,1);break}return r}function U(e,t,n){return this.constructor(e,n)}function A(e,t,n,o,a){e._disable||(e._disable=!0,(e.__ref=t.ref)&&delete t.ref,(e.__key=t.key)&&delete t.key,!e.base||a?e.componentWillMount&&e.componentWillMount():e.componentWillReceiveProps&&e.componentWillReceiveProps(t,o),o&&o!==e.context&&(e.prevContext||(e.prevContext=e.context),e.context=o),e.prevProps||(e.prevProps=e.props),e.props=t,e._disable=!1,0!==n&&(1!==n&&!1===r.syncComponentUpdates&&e.base?f(e):E(e,1,a)),e.__ref&&e.__ref(e))}function E(e,t,n,o){if(!e._disable){var a,i,u,s=e.props,c=e.state,p=e.context,f=e.prevProps||s,d=e.prevState||c,v=e.prevContext||p,m=e.base,h=e.nextBase,y=m||h,b=e._component,C=!1;if(m&&(e.props=f,e.state=d,e.context=v,2!==t&&e.shouldComponentUpdate&&!1===e.shouldComponentUpdate(s,c,p)?C=!0:e.componentWillUpdate&&e.componentWillUpdate(s,c,p),e.props=s,e.state=c,e.context=p),e.prevProps=e.prevState=e.prevContext=e.nextBase=null,e._dirty=!1,!C){a=e.render(s,c,p),e.getChildContext&&(p=l(l({},p),e.getChildContext()));var N,k,P=a&&a.nodeName;if("function"==typeof P){var M=_(a);(i=b)&&i.constructor===P&&M.key==i.__key?A(i,M,1,p,!1):(N=i,e._component=i=T(P,M,p),i.nextBase=i.nextBase||h,i._parentComponent=e,A(i,M,0,p,!1),E(i,1,n,!0)),k=i.base}else u=y,(N=b)&&(u=e._component=null),(y||1===t)&&(u&&(u._component=null),k=S(u,a,p,n||!m,y&&y.parentNode,!0));if(y&&k!==y&&i!==b){var U=y.parentNode;U&&k!==U&&(U.replaceChild(k,y),N||(y._component=null,j(y,!1)))}if(N&&O(N),e.base=k,k&&!o){for(var B=e,L=e;L=L._parentComponent;)(B=L).base=k;k._component=B,k._componentConstructor=B.constructor}}if(!m||n?g.unshift(e):C||(e.componentDidUpdate&&e.componentDidUpdate(f,d,v),r.afterUpdate&&r.afterUpdate(e)),null!=e._renderCallbacks)for(;e._renderCallbacks.length;)e._renderCallbacks.pop().call(e);x||o||w()}}function O(e){r.beforeUnmount&&r.beforeUnmount(e);var t=e.base;e._disable=!0,e.componentWillUnmount&&e.componentWillUnmount(),e.base=null;var n=e._component;n?O(n):t&&(t.__preactattr_&&t.__preactattr_.ref&&t.__preactattr_.ref(null),e.nextBase=t,h(t),function(e){var t=e.constructor.name;(M[t]||(M[t]=[])).push(e)}(e),P(t)),e.__ref&&e.__ref(null)}function B(e,t){this._dirty=!0,this.context=t,this.props=e,this.state=this.state||{}}function L(e,t,n){return S(n,e,{},!1,t,!1)}l(B.prototype,{setState:function(e,t){var n=this.state;this.prevState||(this.prevState=l({},n)),l(n,"function"==typeof e?e(n,this.props):e),t&&(this._renderCallbacks=this._renderCallbacks||[]).push(t),f(this)},forceUpdate:function(e){e&&(this._renderCallbacks=this._renderCallbacks||[]).push(e),E(this,2)},render:function(){}});var W={h:i,createElement:i,cloneElement:s,Component:B,render:L,rerender:d,options:r};t.default=W},"./components/Avatar/Avatar.less":function(e,t){e.exports={avatar:"xENoiNwQovPhb9VWob-vf","avatar--red":"_2SIFrt3wCZqUS_kn7D2Byv","avatar--pink":"Zh2DOTREFnrio_AA1BPmg","avatar--orange":"_1JmeGPYSsra9QT38le1miF","avatar--yellow":"_2MrzK0QVlFmAFYomnBb4CM","avatar--green":"_30P9YE1U8MyfRLj8NfteDu","avatar--grey":"_1OCNleVSd7BbLjNytwBJKj","avatar--blue":"_2RW2-mhuvuNqcY_E2XLxTU",avatarImage:"SVFjb3tTMZrq1rYoyX8W2"}},"./components/Avatar/index.js":function(e,t,n){"use strict";t.__esModule=!0,t.default=void 0;var r=n("../node_modules/preact/dist/preact.esm.js"),o=i(n("./components/Avatar/Avatar.less")),a=i(n("../node_modules/classnames/index.js"));function i(e){return e&&e.__esModule?e:{default:e}}var l=function(e){function t(){return function(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")}(this,t),function(e,t){if(!e)throw new ReferenceError("this hasn't been initialised - super() hasn't been called");return!t||"object"!=typeof t&&"function"!=typeof t?e:t}(this,e.apply(this,arguments))}return function(e,t){if("function"!=typeof t&&null!==t)throw new TypeError("Super expression must either be null or a function, not "+typeof t);e.prototype=Object.create(t&&t.prototype,{constructor:{value:e,enumerable:!1,writable:!0,configurable:!0}}),t&&(Object.setPrototypeOf?Object.setPrototypeOf(e,t):e.__proto__=t)}(t,e),t.prototype.render=function(){var e=this.props,t=e.imageUrl,n=e.name,i=e.color;return(0,r.h)("div",{className:(0,a.default)(o.default.avatar,o.default["avatar--"+i])},(0,r.h)("img",{class:o.default.avatarImage,src:t,alt:n}))},t}(r.Component);t.default=l}});
//# sourceMappingURL=.js.map