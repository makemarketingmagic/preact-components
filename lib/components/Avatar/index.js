'use strict';

exports.__esModule = true;
exports.default = undefined;

var _preact = require('preact');

var _Avatar = require('./Avatar.less');

var _Avatar2 = _interopRequireDefault(_Avatar);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Avatar = function (_Component) {
    _inherits(Avatar, _Component);

    function Avatar() {
        _classCallCheck(this, Avatar);

        return _possibleConstructorReturn(this, _Component.apply(this, arguments));
    }

    Avatar.prototype.render = function render() {
        var _props = this.props,
            imageUrl = _props.imageUrl,
            name = _props.name,
            color = _props.color;

        return (0, _preact.h)(
            'div',
            { className: (0, _classnames2.default)(_Avatar2.default.avatar, _Avatar2.default['avatar--' + color]) },
            (0, _preact.h)('img', { 'class': _Avatar2.default.avatarImage, src: imageUrl, alt: name })
        );
    };

    return Avatar;
}(_preact.Component);

exports.default = Avatar;