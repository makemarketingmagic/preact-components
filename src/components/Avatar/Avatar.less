@import '../common/mixins.less';
@import '../common/variables.less';

.avatar {
    height: 128px;
    width: 128px;
    border: 16px transparent solid;
    overflow: hidden;
    border-radius: 50%;
    .responsive(600px, {
        margin-bottom: 8px;
    });
}

.avatar--red {
    border-color: @red;
}

.avatar--pink {
    border-color: @pink;
}

.avatar--orange {
    border-color: @orange;
}

.avatar--yellow {
    border-color: @yellow;
}

.avatar--green {
    border-color: @green;
}

.avatar--grey {
    border-color: @grey; 
}

.avatar--blue {
    border-color: @blue;
}

.avatarImage {
    height: 100%;
    width: 100%;
}