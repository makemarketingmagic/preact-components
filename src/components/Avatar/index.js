import { h, Component } from 'preact'
import styled from 'styled-components'
import { media, colors } from '../common/scMixins'

const AvatarContainer = styled.div`
    height: 128px;
    width: 128px;
    border: 16px transparent solid;
    overflow: hidden;
    border-radius: 50%;
    border-color: ${(props) => {
        return colors[props.color];
    }};
    ${media.mobile('margin-bottom: 8px;')}
`,
    Image = styled.img`
    height: 100%;
    width: 100%;
`
export default class Avatar extends Component {
    render() {
        const { imageUrl, name, color } = this.props
        return (
            <AvatarContainer color={color}>
                <Image src={imageUrl} alt={name} />
            </AvatarContainer>
        )
    }
}