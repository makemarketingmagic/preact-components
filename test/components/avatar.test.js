import { h } from 'preact';
import render from 'preact-render-to-string';
import Avatar from '../../src/components/Avatar';

describe('Avatar', () => {
    it('should render', () => {
        const element = render(<Avatar
            imageUrl='test.png'
            name='Test Name'
            color='orange'
        />)
        expect(element).toMatchSnapshot()
    })
})