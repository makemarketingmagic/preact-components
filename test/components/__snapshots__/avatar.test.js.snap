// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render 1`] = `
<div class="avatar avatar--orange">
  <img class="avatarImage" src="test.png" alt="Test Name" />
</div>
`;
